src/lib.rs

```diff
@@ -1022,14 +1021,78 @@ trait OperationTrait {
 
 ////////////////////////
 
+/// Returns a string describing version information of the
+/// TensorFlow library. TensorFlow using semantic versioning.
+pub fn version() -> std::result::Result<String, Utf8Error> {
+  unsafe {
+    CStr::from_ptr(tf::TF_Version()).to_str().map(|s| s.to_string())
+  }
+}
+
+////////////////////////
+
+/// A TensorShape is the shape of a tensor.  A TensorShape may be an unknown
+/// rank, or it may have a known rank with each dimension being known or
+/// unknown.
+#[derive(Debug,Eq,Ord,PartialEq,PartialOrd,Hash,Clone)]
+pub struct TensorShape(Option<Vec<Option<i64>>>);
+
+impl TensorShape {
+  /// Returns the number of dimensions if known, or None if unknown.
+  pub fn rank(&self) -> Option<usize> {
```

Comment:
We're generally discouraging the use of the term `rank`:
https://www.tensorflow.org/how_tos/documentation/
`Refer to tensors with 3 or more dimensions as 3-D tensors or n-D tensors. Use the word "rank" only if it makes sense, but try to use "dimension" instead. Never use the word "order" to describe the size of a tensor.`

In the C++ class, we're using the convention dims() and dim_size().
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor_shape.h